src/Component/CaptionedSlide.php

```diff
@@ -25,30 +25,30 @@ final class CaptionedSlide implements HasCaption {
 	private $slide_node;
 
 	/**
-	 * The caption text.
+	 * The caption node.
 	 *
-	 * @var string
+	 * @var DOMNode
 	 */
-	private $caption;
+	private $caption_node;
 
 	/**
 	 * Constructs the class.
 	 *
-	 * @param DOMNode $slide_node The slide node, possibly an image.
-	 * @param string  $caption    The caption text.
+	 * @param DOMNode $slide_node   The slide node, possibly an image.
+	 * @param DOMNode $caption_node The caption node.
 	 */
-	public function __construct( DOMNode $slide_node, $caption ) {
-		$this->slide_node = $slide_node;
-		$this->caption    = $caption;
+	public function __construct( DOMNode $slide_node, DOMNode $caption_node ) {
+		$this->slide_node   = $slide_node;
+		$this->caption_node = $caption_node;
 	}
 
 	/**
-	 * Gets the caption text.
+	 * Gets the caption node.
 	 *
-	 * @return string The caption text.
+	 * @return DOMNode
 	 */
-	public function get_caption() {
-		return $this->caption;
+	public function get_caption_node() {
+		return $this->caption_node;
 	}
```

Comment:
Isn't the `$caption_node` strictly only ever a `DOMElement` while the plugin runs and not a `DOMNode`? I see it is tested as a `DOMText` but from what I can see the only value provided here is the response of one of the two `get_caption_element` methods which is _either_ `DOMElement` or `null`.

Reply:
Ah yes that's true. Before the refactor in https://github.com/ampproject/amp-wp/pull/4775/commits/19b45c96cb08b9600acad9c91f44bcd45579405e it was possibly for the caption to be text only (which would eventually become `DOMText`), but that's no longer the case.

Reply:
Resolved in https://github.com/ampproject/amp-wp/pull/4775/commits/0bd44f9f1dc40c136c6c276c19cef29ba912c114.